lxd/util/sys.go

```diff
@@ -110,3 +110,14 @@ func AddFileDescriptor(fdFiles *[]*os.File, file *os.File) int {
 	*fdFiles = append(*fdFiles, file)
 	return 2 + len(*fdFiles) // Use 2+fdFiles count, as first user file descriptor is 3.
 }
+
+// ShortenedSocketPath creates a shorter alternative path to a socket by using the file descriptor to the directory of the socket file.
+// Used to handle paths > 108 chars.
+func ShortenedSocketPath(originalSockPath string, fdFiles *[]*os.File) (string, error) {
+	configSockDir, err := os.Open(filepath.Dir(originalSockPath)) // Files opened here must be closed outside this function once they are not needed anymore.
```

Comment:
The comment on this line should be in the function description so that the caller doesnt need to read the implementation of the function to know this.